src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteWalletByIdTest.java

```diff
@@ -0,0 +1,68 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import pl.byczazagroda.trackexpensesappbackend.BaseIntegrationTestIT;
+import pl.byczazagroda.trackexpensesappbackend.exception.ErrorCode;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransaction;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.model.Wallet;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionRepository;
+import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+
+class DeleteWalletByIdTest extends BaseIntegrationTestIT {
+
+    @Autowired
+    WalletRepository walletRepository;
+
+    @Autowired
+    FinancialTransactionRepository financialTransactionRepository;
+
+    @BeforeEach
+    void clearDatabase() {
+        walletRepository.deleteAll();
+    }
+
+    @DisplayName("Should delete wallet from a database and return status 'OK'")
+    @Test
+    void testDeleteWalletByIdAPI_whenWalletIdIsCorrect_thenShouldReturnAcceptAndDeleteRecord () throws Exception {
+        Wallet wallet = walletRepository.save(new Wallet("Test Wallet"));
+        createTestFinancialTransaction(wallet);
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/wallets/{id}", wallet.getId()).accept(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+        Assertions.assertEquals(0, walletRepository.count());
+    }
+
+    @DisplayName("Should return is Not Found error when Id does not exist in a database")
+    @Test
+    void testDeleteWalletById_whenWalletIdIsIncorrect_thenShouldReturnNotFoundError() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/wallets/{id}", 1L).accept(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isNotFound())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.status").value(ErrorCode.W003.getBusinessStatus()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.message").value(ErrorCode.W003.getBusinessMessage()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.statusCode").value(ErrorCode.W003.getBusinessStatusCode()));
+        Assertions.assertEquals(0, walletRepository.count());
+    }
+
+    private void createTestFinancialTransaction(Wallet wallet){
+        financialTransactionRepository.save(FinancialTransaction.builder()
+                .wallet(wallet)
+                .amount(new BigDecimal("2.0"))
+                .date(Instant.ofEpochMilli(0L))
```

Comment:
Locally, this test in the application passes for a value of 0, because in the database the date is stored with an offset of 1 hour. On github, this offset is not present and therefore we get an exception.

@mirekgab conducted the test and received the following results:
In the BaseIntegrationTestIT.java class, we create a mySQLContainer object but do not specify the server time zone in it.

In the test class we have a method createTestFinancialTransaction. In it, he set the time to 1 second. Checked the data in the mysql database and got these results:

1. test containers without setting the time zone
+----+-----------+------------------+--------+---------------------+-------------+
| id | wallet_id | transaction_type | amount | transaction_date | description |
+----+-----------+------------------+--------+---------------------+-------------+
| 1 | 1 | INCOME | 5.00 | 1970-01-01 01:00:01 | hello |
+----+-----------+------------------+--------+---------------------+-------------+

2. test containers with time zone set to UTC
+----+-----------+------------------+--------+---------------------+-------------+
| id | wallet_id | transaction_type | amount | transaction_date | description |
+----+-----------+------------------+--------+---------------------+-------------+
| 1 | 1 | INCOME | 5.00 | 1970-01-01 00:00:01 | hello |
+----+-----------+------------------+--------+---------------------+-------------+

Please set date to Instant.ofEpochSecond(1L)